platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1363,6 +1392,32 @@ - (id)mgl_jsonHasExpressionObject {
     return expressionObject;
 }
 
+- (id)mgl_jsonFormatExpressionObject {
+    BOOL isAftermarketFunction = [self.function hasPrefix:@"mgl_attributed:"];
+    NSExpression *formatParameter;
+    NSArray<MGLAttributedExpression *> *attributedExpressions;
+    
+    if (isAftermarketFunction) {
+        formatParameter = self.arguments.firstObject;
+    } else {
+        formatParameter = self.operand;
+    }
+    
+    if ([formatParameter respondsToSelector:@selector(constantValue)] && [formatParameter.constantValue isKindOfClass:[NSArray class]]) {
```

Comment:
A developer might be tempted to create a constant-value NSExpression from a bare MGLAttributedExpression, which would be an error here. Maybe we could convert a standalone, constant MGLAttributedExpression into `format` expression too?

Reply:
If we do that, then it starts to seem like we could combine this operator with `mgl_join:` so that joining MGLAttributedExpressions together creates a `format` expression instead of a `concat` expression.